language/stdlib/modules/LibraAccount.move

```diff
@@ -857,10 +867,27 @@ module LibraAccount {
         txn_gas_price: u64,
         txn_max_gas_units: u64,
         txn_expiration_time: u64,
-        chain_id: u8
+        chain_id: u8,
+        transaction_type: u8,
```

Comment:
Do we need this? It seems like we could get away with something like:

```
if (script_hash.is_empty() {
  // module tx logic
} else {
  // script tx logic
}
```

That will let us get rid of the `EPROLOGUE_UNEXPECTED_WRITESET` error code and the `transaction_type` parameter (both here and in the Rust `TransactionMetadata`).

Reply:
My concern is if we add more types to TransactionPayload in the future. Having an explicit tag here allow us to have different prologue logic for each payload type in the future. What do you think?